internal/dataplane/parser/translate_secrets_test.go

```diff
@@ -4,14 +4,14 @@ import (
 	"testing"
 
 	"github.com/kong/go-kong/kong"
-	"github.com/stretchr/testify/require"
 
 	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/kongstate"
 )
 
 func TestGetPluginsAssociatedWithCACertSecret(t *testing.T) {
 	secretID := "8a3753e0-093b-43d9-9d39-27985c987d92" //nolint:gosec
-	plugins := []kongstate.Plugin{
+	// todo: adapt to implementation
+	_ = []kongstate.Plugin{
```

Comment:
Given the PR is already marked as ready to review, I believe this needs addressing.

Reply:
Please note it's marked as ready for review (as Michał has suggested doing), but still is not meant to be merged to main in this form. I can adapt the tests but IMO it's not worth it for now until we decide that we wanna go with this approach. The key output I wanted to get out of reviewing this is whether the team thinks it's the way we wanna go regarding events publishing (mainly, if it's ok to do that directly from Parser).